uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Service/WorkPatternTest.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+use CRM_HRLeaveAndAbsences_BAO_WorkPattern as WorkPattern;
+use CRM_HRLeaveAndAbsences_Test_Fabricator_WorkPattern as WorkPatternFabricator;
+use CRM_HRLeaveAndAbsences_Test_Fabricator_ContactWorkPattern as ContactWorkPatternFabricator;
+use CRM_HRLeaveAndAbsences_Service_WorkPattern as WorkPatternService;
+
+
+/**
+ * Class CRM_HRLeaveAndAbsences_Service_WorkPatternTest
+ *
+ * @group headless
+ */
+class CRM_HRLeaveAndAbsences_Service_WorkPatternTest extends BaseHeadlessTest {
+
+  private $workPatternService;
+
+  public function setUp() {
+    CRM_Core_DAO::executeQuery('SET foreign_key_checks = 0;');
+    $this->workPatternService = new WorkPatternService();
+  }
+
+  public function tearDown() {
+    CRM_Core_DAO::executeQuery('SET foreign_key_checks = 1;');
+  }
+
+  public function testWorkPatternHasEverBeenUsedReturnsTrueWhenWorkPatternIsTheDefaultWorkPattern() {
+    $workPattern = WorkPatternFabricator::fabricate(['is_default' => 1]);
+    $this->assertTrue($this->workPatternService->workPatternHasEverBeenUsed($workPattern->id));
+  }
+
+  public function testWorkPatternHasEverBeenUsedReturnsTrueWhenWorkPatternIsLinkedToAContactWorkPattern() {
```

Comment:
Given the method name change I suggested, you'll probably want to change this related test name as well